tsdb_k8s_queries/README.md

```diff
@@ -0,0 +1,91 @@
+## TSDB k8s query track
+
+The main goal of the TSDB k8s query track is to measure the performance of common k8s integration search requests.
+The queries, index templates and corpus data try to match production as close as possible.
+
+### The search requests
+
+The search requests of the following visualizations have been included in this track:
+* Average CPU usage per pod.
+* Average memory usage per pod.
+* Average cpu per container.
+* Average memory per container.
+* Average node cpu usage per container.
+* Average node memory usage per container.
+* Unique deployment count.
+* Percentile cpu usage per container.
+* Status per pod.
+* Transmitted network usage per pod.
+
+The search request is sourced from Kibana Lens in the k8s integration. For each of this search request a template is defined in operations.
+From each template, three search requests are generated. First with the last 15 minutes filter, then the last 2 hours filter, and finally with a last 24-hour interval. In case a `date_histogram` is used in the query template then each of these variations uses a different interval. A fixed interval of 30 seconds, 1 minute, and 30 minutes respectively.
+
+The k8s visualizations that run these queries don't run very often or under a high query load.
+Often these visualizations are loaded and then sometime later re-loaded. This triggers the shards of the k8s pod and container data streams to go search-idle. However, indexing always happens in the background. When shards become search-active again, a refresh needs to occur as part of the search request. This track is designed to emulate this runtime behaviour.
+
+The way that this is emulated is by executing search and bulk operations concurrently. For each of the mentioned visualization there is a search operation and a corresponding bulk operation with a `touch-*` prefix as name. Both of the search and bulk operation are run concurrently using a `parallel` task. The bulk operations perform minimal indexing (1 index client and one document per bulk request) and reindexes the same curpus with a different `@timestamp` field using an ingest processor. This is sufficient so that the refresh isn't a noop. Also the data streams are configured with a `index.search.idle.after` of `1s`, so that shards of the backing indices become search-idle after 1 second of the last search request. Additionally, the search operations are configured to only execute a search every 4 seconds.
```

Comment:
```suggestion
The way that this is emulated is by executing search and bulk operations concurrently. For each of the mentioned visualization there is a search operation and a corresponding bulk operation with a `touch-*` prefix as name. Both of the search and bulk operation are run concurrently using a `parallel` task. The bulk operations perform minimal indexing (1 index client and one document per bulk request) and reindexes the same corpus with a different `@timestamp` field using an ingest processor. This is sufficient so that the refresh isn't a noop. Also the data streams are configured with a `index.search.idle.after` of `1s`, so that shards of the backing indices become search-idle after 1 second of the last search request. Additionally, the search operations are configured to only execute a search every 4 seconds.
```